spring-integration-mongodb/src/test/java/org/springframework/integration/mongodb/test/entity/QueryMessage.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2007-2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.mongodb.test.entity;
+
+/**
+ * @author Xavier Padró
+ * @since 5.0
+ */
+public class QueryMessage {
```

Comment:
M-m-m. Seems for me we can live with this class. 
If you would like to check if expression works well, maybe try to perform it against some header, e.g.:
```
.queryExpression("headers.query")
```